include/z64.h

```diff
@@ -615,227 +624,257 @@ typedef struct {
-    /* 0xC3 */ u8 unkC3;
-    /* 0xC4 */ Color_RGB8 unkC4;
-    /* 0xC7 */ s8 unkC7;
-    /* 0xC8 */ s8 unkC8;
-    /* 0xC9 */ s8 unkC9;
-    /* 0xCA */ Color_RGB8 unkCA;
-    /* 0xCD */ s8 unkCD;
-    /* 0xCE */ s8 unkCE;
-    /* 0xCF */ s8 unkCF;
-    /* 0xD0 */ Color_RGB8 unkD0;
-    /* 0xD3 */ Color_RGB8 unkD3;
-    /* 0xD6 */ s16 unkD6;
-    /* 0xD8 */ s16 unkD8;
-    /* 0xDA */ UNK_TYPE1 unkDA;
-    /* 0xDB */ UNK_TYPE1 unkDB;
-    /* 0xDC */ f32 unkDC;
-    /* 0xE0 */ u8 unkE0;
-    /* 0xE1 */ UNK_TYPE1 unkE1;
-    /* 0xE2 */ s8 unkE2;
-    /* 0xE3 */ u8 unkE3; // modified by unused func in EnWeatherTag
-    /* 0xE4 */ UNK_TYPE1 unkE4;
-    /* 0xE5 */ UNK_TYPE1 unkE5;
-    /* 0xE6 */ UNK_TYPE1 unkE6;
-    /* 0xE7 */ UNK_TYPE1 unkE7;
-    /* 0xE8 */ UNK_TYPE1 unkE8;
-    /* 0xE9 */ UNK_TYPE1 unkE9;
-    /* 0xEA */ u8 unkEA;
-    /* 0xEB */ UNK_TYPE1 unkEB;
-    /* 0xEC */ UNK_TYPE1 unkEC;
-    /* 0xED */ UNK_TYPE1 unkED;
-    /* 0xEE */ UNK_TYPE1 unkEE;
-    /* 0xEF */ UNK_TYPE1 unkEF;
-    /* 0xF0 */ UNK_TYPE1 unkF0;
-    /* 0xF1 */ UNK_TYPE1 unkF1;
-    /* 0xF2 */ u8 unkF2[0xC]; // F2-F6 are used by weather tag
-} KankyoContext; // size = 0xFE
+typedef struct {
+    /* 0x00 */ u16 unk_0;
+    /* 0x02 */ u16 unk_2;
+    /* 0x04 */ f32 unk_4;
+    /* 0x08 */ f32 unk_8;
+    /* 0x0C */ f32 unk_C;
+    /* 0x10 */ u8 unk_10;
+    /* 0x11 */ u8 unk_11;
+    /* 0x12 */ u8 unk_12;
+    /* 0x13 */ u8 unk_13;
+    /* 0x14 */ u8 unk_14;
+    /* 0x15 */ u8 unk_15;
+    /* 0x16 */ u8 unk_16;
+    /* 0x17 */ u8 unk_17;
+    /* 0x18 */ u8 unk_18;
+    /* 0x19 */ u8 unk_19;
+    /* 0x1A */ u16 unk_1A;
+    /* 0x1C */ u16 unk_1C;
+    /* 0x1E */ u8 unk_1E;
+    /* 0x1F */ u8 unk_1F;
+    /* 0x20 */ u8 unk_20;
+    /* 0x21 */ u8 unk_21;
+    /* 0x22 */ u16 unk_22;
+    /* 0x24 */ u16 unk_24;
+    /* 0x26 */ u8 unk_26;
+    /* 0x28 */ LightInfo unk_28; // sun 1
+    /* 0x36 */ LightInfo unk_36; // sun 2
+    /* 0x44 */ s8 unk_44;
+    /* 0x48 */ DmaRequest unk_48;
+    /* 0x68 */ OSMesgQueue unk_68;
+    /* 0x80 */ OSMesg unk_80;
+    /* 0x84 */ f32 unk_84;
+    /* 0x88 */ f32 unk_88;
+    /* 0x8C */ LightSettings2 unk_8C;
+    /* 0xA8 */ f32 unk_A8;
+    /* 0xAC */ Vec3s windDir;
+    /* 0xB4 */ f32 windSpeed;
+    /* 0xB8 */ u8 nbLightSettings;
+    /* 0xBC */ LightSettings* lightSettingsList;
+    /* 0xC0 */ u8 unk_C0;
+    /* 0xC1 */ u8 unk_C1;
+    /* 0xC2 */ u8 unk_C2;
+    /* 0xC3 */ u8 unk_C3;
+    /* 0xC4 */ LightSettings unk_C4;
+    /* 0xDA */ u16 unk_DA;
+    /* 0xDC */ f32 unk_DC;
+    /* 0xE0 */ u8 unk_E0;
+    /* 0xE1 */ u8 unk_E1;
+    /* 0xE2 */ s8 unk_E2;
+    /* 0xE3 */ u8 unk_E3; // modified by unused func in EnWeatherTag
+    /* 0xE4 */ u8 unk_E4;
+    /* 0xE5 */ u8 unk_E5;
+    /* 0xE6 */ u8 unk_E6[4];
+    /* 0xEA */ u8 unk_EA;
+    /* 0xEB */ u8 unk_EB;
+    /* 0xEC */ u8 unk_EC;
+    /* 0xED */ u8 unk_ED;
+    /* 0xEE */ u8 unk_EE[4];
+    /* 0xF2 */ u8 unk_F2[8];
+    /* 0xFA */ u8 unk_FA[4];
+} EnvironmentContext; // size = 0xFE
```

Comment:
```suggestion
} EnvironmentContext; // size = 0x100
```
This structure contains 32-bit wide types so will pad to the next multiple of 0x4.